### DISCUSSION TITLE
A simple way to add more protocols for torrents.

### DISCUSSION THREAD

**Discussion by master255** [arvidn/libtorrent#7782] [Ideas]
@arvidn It is very difficult to connect the operation of different network systems.
This is further complicated by different programming languages and of course data transfer protocols.
Therefore, we need a single standard for information transfer between systems.
I think this standard should be the http protocol!
It is the shortest, most common, and simplest. Everyone knows how get requests are organized.

Therefore, further protocol additions should be made by adding a web server to libtorrent that can retrieve data and error codes from third-party systems.
On the libtorrent side, it's necessary to improve the url-seed handling system. To make the library understand http error codes better.
We need an example of a local web server to add our own protocols.
All it's necessary for the programmer to do is to add data transfer functions to the web server. From any other system.
This way it is easier, faster and more efficient to add any protocol to torrents.

**Comment by arvidn** [comment]
you can also use `add_piece()`. https://libtorrent.org/reference-Torrent_Handle.html#add-piece

**Reply by master255** [reply]
Yes. But it can run slower because there is no data transfer buffer optimization as in sockets. And there is performance degradation due to C++ function call in my Java program. So it's not really an efficient way to do it.
